libprio.i

```diff
@@ -137,39 +139,101 @@ OPAQUE_POINTER(PrivateKey)
 
 // PrioVerifier_set_data
 %typemap(in) (unsigned char * data, unsigned int dataLen) {
-    if (!PyByteArray_Check($input)) {
-        PyErr_SetString(PyExc_ValueError, "Expecting a bytearray");
+    if (!PyBytes_Check($input)) {
+        PyErr_SetString(PyExc_ValueError, "Expecting a byte string");
         SWIG_fail;
     }
-    $1 = (unsigned char*) PyByteArray_AsString($input);
-    $2 = (unsigned int) PyByteArray_Size($input);
+    $1 = (unsigned char*) PyBytes_AsString($input);
+    $2 = (unsigned int) PyBytes_Size($input);
 }
 
 
 // PrioTotalShare_final
 %typemap(in) (const_PrioConfig, unsigned long *) {
-    $1 = (const_PrioConfig)PyLong_AsVoidPtr($input);
-    $2 = (unsigned long*) malloc(sizeof(long)*PrioConfig_numDataFields($1));
+    $1 = PyCapsule_GetPointer($input, "PrioConfig");
+    $2 = malloc(sizeof(long)*PrioConfig_numDataFields($1));
 }
 
 %typemap(argout) (const_PrioConfig, unsigned long *) {
     $result = SWIG_Python_AppendOutput(
         $result,
         PyByteArray_FromStringAndSize((const char*)$2, sizeof(long)*PrioConfig_numDataFields($1))
     );
-    free($2);
+    if ($2) free($2);
 }
 
 %apply (const_PrioConfig, unsigned long *) {
     (const_PrioConfig cfg, unsigned long *output)
 }
 
 
+// Define wrapper functions for msgpacker_packer due to the dynamically allocated sbuffer.
+// Memory shouldn't be moving around so much, but this wrapper function needs to exist in
+// order to marshal data due to typemaps not having side-effects
+
+%define MSGPACK_WRITE_WRAPPER(type)
+%inline %{
+PyObject* type ## _write_wrapper(const_ ## type p) {
+    PyObject* data = NULL;
+    msgpack_sbuffer sbuf;
+    msgpack_packer pk;
+    msgpack_sbuffer_init(&sbuf);
+    msgpack_packer_init(&pk, &sbuf, msgpack_sbuffer_write);
+
+    SECStatus rv = type ##_write(p, &pk);
+    if (rv == SECSuccess) {
+        // move the data outside of this wrapper
+        data = PyBytes_FromStringAndSize(sbuf.data, sbuf.size);
+    }
+
+    // free msgpacker data-structures
+    msgpack_sbuffer_destroy(&sbuf);
+
+    return data;
+}
+%}
+%enddef
+
+MSGPACK_WRITE_WRAPPER(PrioPacketVerify1)
+MSGPACK_WRITE_WRAPPER(PrioPacketVerify2)
+MSGPACK_WRITE_WRAPPER(PrioTotalShare)
+
+
+// Unfortunately, unpacking also requires some manual work. We take the binary message and
+// copy the data into an unpacker
+
+%apply (const unsigned char*, unsigned int) {
+    (const unsigned char *data, unsigned int len)
+}
+
+%define MSGPACK_READ_WRAPPER(type)
+%inline %{
+SECStatus type ## _read_wrapper(type p, const unsigned char *data, unsigned int len, const_PrioConfig cfg) {
+    SECStatus rv = SECFailure;
+    msgpack_unpacker upk;
+    bool result = msgpack_unpacker_init(&upk, len+1);
```

Comment:
This line caused a lot of issues, apparently there needs to be extra padding to prevent overwriting bounds.